rfcs/0013-redesign-assets-id-system.md

```diff
@@ -0,0 +1,246 @@
+to conduct an initial release using a simplistic storage migration to create
+initial pools for Pablo. However, with the dependence on these hard-coded asset
+IDs - we now must conduct additional, more complex, storage migrations to add 
+more Pablo pools and more tokens.
+
+Instead of depending on these hard-coded asset IDs, we should instead develop
+a dynamic ID system that makes it easy for developers to both interact with 
+assets within our consensus system and to retrieve information relevant to a 
+given asset.
+
+## Requirements
+
+* There exists a double mapping between multi-locations and local asset IDs
+  * Zero or One multi-locations ↔ One asset ID
+
+* Given a way to uniquely identify an asset, one should be able to retrieve 
+asset metadata (ticker-symbol, decimal precision, and ratio)
+
+* Support the traits from `frame_support::traits::tokens::fungibles`
+  * Specifically `MutateHold`
+
+* Ability to mint mintable assets and the inability to mint non-mintable assets
+
+* Support the existing local asset IDs we have hard-coded
+
+* Utilize both pallet-balances and pallet-assets from Parity for maintaining 
+balances within our consensus system
+
+# Other Solutions
+
+<!-- TODO: These are mostly done, but I will introduce them to the repo in 
+their own files -->
+
+# Design
+
+Given the scope of assets, the design will be broken down into several more 
+consumable sections.
+
+## Local Asset ID Generation
+
+Each asset within our consensus system should have its own local asset ID. 
+Currently, we have hard-coded these local IDs rather than dynamically generating 
+them as needed. While existing assets should maintain their current ID, we can 
+define a new way to derive asset IDs automatically for assets created in the 
+future.
+
+A simple approach to this would be processing a nonce using a fixed-length 
+hashing algorithm. Given the fixed length of the hash output, no collisions will
+exist between previous hard-coded asset IDs and ones generated from the hash.
+
+## Asset Types and Asset Routing
+
+Within our consensus system, we have two primary types of currency mintable and 
+non-mintable. Mintable assets tend to be purely local in their scope and 
+function while non-mintable assets tend to come into our consensus system 
+externally (via XCM). To ensure no overlap in function between these assets,
+they each should belong to their own instance of pallet-assets.
+
+When also considering pallet-balances for our native asset, it becomes clear 
+that some amount of assisted routing will be necessary. While routing between 
+pallet-assets and pallet-balances is trivial, routing between pallet-balances 
+and two instances of pallet-assets becomes more difficult.
+
+Ideally, this routing would not be needed, but both our local and foreign asset 
+will, in some instances, interact within the same scope (i.e. Pablo).
+
+Our current asset router (also called pallet-assets, to be renamed to assets 
```

Comment:
I think this routing possibly is not needed if we correctly follow a namespacing approach such as multi-asset to identify the pallet and/or instance the asset belongs to. Then the routing becomes just a library of code rather than a fully fledged pallet. Then the purpose of asset-registry would just be exposing any additional globabl meta data such as fees, symbol, ratios etc.

Reply:
Let me be more precise. Suggestion is not to remove the routing but to avoid the assets-registry in making the routing decisions. Given the multiaddress of the multiasset already carries the pallet instance information, whether it is mintable asset or not is a straightforward logic decision with static runtime information.

Reply:
I see, that's more clear. Thank you.
I'll revise the asset ID generation section with this in mind and update the docs accordingly.